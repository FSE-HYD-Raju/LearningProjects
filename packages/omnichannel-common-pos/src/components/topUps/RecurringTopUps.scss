.RecurringTopUps {
	.w-view-header {
		align-items: center;
	}
	h3 {
		font-weight: 600;
		span {
			text-transform: uppercase;
		}
	}
	table {
		th,
		td {
			border: 0;
			vertical-align: top;
		}
	}

	&-mobile-view-payment-method {
		width: 7.6rem;
		text-align: right;
	}

	&-right-align {
		text-align: right;
	}

	&-center-align {
		text-align: center;
	}

	&-active-span {
		cursor: pointer;
		color: $primary;
	}

	&-key-value {
		justify-content: space-between;
	}

	&-payment-method-col {
		& > *:first-child {
			margin-left: 0;
		}
		i::before {
			margin-left: 0;
		}
	}

	&-add-button {
		margin-top: 1rem;
		width: 100%;
	}

	&-fake-btn-link {
		display: flex;
		align-items: flex-start;
		justify-content: flex-start;
	}

	&-add-payment-method-icon {
		font-size: 1.5rem;
		color: $danger;
	}

	&-add-payment-method-link {
		color: $danger;
		text-decoration: underline;
		cursor: pointer;
	}

	&-empty-list {
		display: flex;
		margin-top: 2.375rem;
		margin-bottom: 6rem;
		&-spacer {
			flex: 1 0 0;
		}
		&-content {
			flex: 0 1 30rem;
			* {
				display: block;
				text-align: center;
			}
		}
		&-title-icon {
			margin-bottom: 0.875rem;
			font-size: 5rem;
			color: #eeeeee;
		}
		&-description {
			margin-bottom: 1.25rem;
		}
		&-btn-outline-primary {
			margin: 0 auto;
		}
	}

	@include media-breakpoint-up(sm) {
		&-add-button {
			margin: 0 auto;
			padding: 0 1.25rem;
			width: initial;
		}
		header {
			h3 {
				span {
					text-transform: none;
				}
			}
			button {
				margin: 0;
			}
		}
		table {
			th,
			td {
				padding: 0.875rem 0.5rem;
			}
			tbody {
				tr:nth-child(odd) {
					background: $light-grey-lighter;
				}
			}
			tr th:nth-child(3),
			tr td:nth-child(3) {
				text-align: right;
			}
			tr th:nth-child(4),
			tr td:nth-child(4) {
				text-align: right;
			}
			tr td:first-child,
			tr th:first-child {
				padding-left: 1.5rem;
			}
			tr td:last-child,
			tr th:last-child {
				padding-right: 1.5rem;
			}
			thead {
				&-edit-column {
					width: 1%;
					min-width: 7rem;
				}
				&-delete-column {
					width: 1%;
					min-width: 5rem;
				}
			}
		}

		&-key {
			display: none;
		}

		&-mobile-view-payment-method {
			width: initial;
			text-align: initial;
		}

		&-payment-method-col {
			& > *:first-child {
				margin-left: 3rem;
			}
			i::before {
				margin-left: -0.3rem;
			}
		}
	}

	@include media-breakpoint-only(xs) {
		thead {
			display: none;
		}
		tr {
			background-color: $white;
			display: flex;
			flex-flow: row wrap;
			padding: 0.5rem 1rem;
			margin-bottom: 0.5rem;
		}
		tr:last-child {
			margin-bottom: 0;
		}
		td {
			flex: 1 100%;
			padding: 0.25rem 0;
			display: flex;
			flex-direction: row;
			justify-content: space-between;
		}
		header {
			display: flex;
			flex-flow: row wrap;

			h3 {
				flex: 1 100%;
			}
		}
		&-fake-btn-link {
			justify-content: flex-end;
		}
		&-add-payment-method-link {
			width: min-content;
		}
	}
}
