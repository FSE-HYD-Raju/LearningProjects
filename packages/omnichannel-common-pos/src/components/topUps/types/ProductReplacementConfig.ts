interface ProductReplacementConfig {
	individualId?: string;
	agreementId?: string;
	productId?: string;
	productOfferingId?: string;
	replaceType?: string;
	configuration?: object;
}
export { ProductReplacementConfig };
