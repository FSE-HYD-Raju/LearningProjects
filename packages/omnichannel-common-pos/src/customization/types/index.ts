import { CustomizationBootstrapType, GetStoreConfig } from "./CustomizationBootstrapType";
import { CustomizationClass } from "./CustomizationClass";
import { CustomizationFunction } from "./CustomizationFunction";
import { CustomizationPointsMapping } from "./CustomizationPointsMapping";

export {
	GetStoreConfig,
	CustomizationBootstrapType,
	CustomizationClass,
	CustomizationFunction,
	CustomizationPointsMapping
};
