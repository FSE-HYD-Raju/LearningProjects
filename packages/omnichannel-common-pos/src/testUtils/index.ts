import ReactWidgetsTestUtils from "./ReactWidgetsTestUtils";
import * as MockDataMaker from "./mock";
const {
	shallowWithContextAndRouterProps,
	shallowWithContextAndLifecycle,
	deepShallowWithContextAndLifecycle,
	deepShallowWithContext,
	shallowWithContext,
	mountWithContextAndRouterProps,
	mountWithContext,
	attachWithContext,
	detach
} = require("./custom-mounters");
const TestUtils = require("./TestUtils");
const MockData = require("./MockData");
const SimpleDataMock = require("./mock/simpleDataMock");

export {
	MockData,
	ReactWidgetsTestUtils,
	TestUtils,
	SimpleDataMock,
	shallowWithContextAndRouterProps,
	shallowWithContextAndLifecycle,
	deepShallowWithContextAndLifecycle,
	deepShallowWithContext,
	shallowWithContext,
	mountWithContextAndRouterProps,
	mountWithContext,
	attachWithContext,
	detach,
	MockDataMaker
};

export * from "./MockRouterProps";

export default TestUtils;
