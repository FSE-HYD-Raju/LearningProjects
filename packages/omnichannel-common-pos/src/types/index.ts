import { ContextType, contextTypesValidationMap } from "./ContextType";

export {
	ContextType,
	contextTypesValidationMap
};
