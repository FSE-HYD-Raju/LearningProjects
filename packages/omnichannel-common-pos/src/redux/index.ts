"use strict";

import { default as actions, AppActions } from "./actions";
import { default as reducers, AppState } from "./reducers";
import { InterceptReducers } from "./reducers/reducerUtils";

import { default as DevTools } from "./DevTools";

import { connectFluxRedux, getDefaultFluxStoreMap } from "./connectFluxRedux";
import { configureStore, AppStore, StoreConfig } from "./store";
import * as actionTypes from "./actions";
import * as decorators from "./decorators";

import * as channelUtils from "./utils/channel";
export * from "./utils";
import * as Selectors from "./selectors";

import { initialState } from "./reducers/initialState";
import BaseService from "./services/BaseService";
export * from "./settings/core";
import { productToBasketItem as productToBasketItemTypes } from "./models/feature/feature.types";
// uses "configureStore" from ./store, should be imported after ./store
import * as ReduxTestUtils from "./utils/ReduxTestUtils";

export {
	decorators,
	Selectors,
	actions,
	reducers,
	DevTools,
	StoreConfig,
	InterceptReducers,
	AppStore,
	initialState,
	AppActions,
	AppState,
	connectFluxRedux,
	configureStore,
	getDefaultFluxStoreMap,
	ReduxTestUtils,
	channelUtils,
	actionTypes,
	BaseService,
	productToBasketItemTypes,
};

export * from "./actions/actionsPayload";
export * from "./services/Rest";
export * from "./services";
export * from "./types";
export * from "./decorators";
export * from "./models/msisdnSelection/msisdnSelection.actions";
export * from "./models/productOfferingConfiguration/productOfferingConfiguration.utils";
