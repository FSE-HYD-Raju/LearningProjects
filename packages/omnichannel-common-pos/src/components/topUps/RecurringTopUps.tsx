import messages from "./RecurringTopUp.messages";
import { Alert } from "react-bootstrap";
import { Component, ReactNode } from "react";
import { RecurringTopUpUtil } from "./RecurringTopUpUtil";
import OcCurrency from "../ocComponents/OcCurrency";
import cssns from "../../utils/cssnsConfig";
import { CustomerPaymentMethod, RecurringTopUpModelType, User } from "../../redux";
import { ContextType, contextTypesValidationMap } from "../../types";
import FormattedMessage, { FormattedMessageDescriptor } from "../../channelUtils/FormattedMessage";
import { FormattedHTMLMessage } from "../../channelUtils";
import { default as RecurringTopUpAddModal } from "./RecurringTopUpAddModal";
import RecurringTopUpEditModal from "./RecurringTopUpEditModal";
import { RecurringTopUpProductType } from "../../redux/types";
import MsisdnUtil from "../../utils/MsisdnUtil";
import PaymentUtil from "../../utils/PaymentUtil";
import { commonDigitalLifeRoutes } from "../../routes/commonRoutesMap";
import { OcButton, OcButtonType } from "../ocComponents";
import RecurringTopUpsEmptyList from "./RecurringTopUpsEmptyList";
import classnames from "classnames";
import RecurringTopUpNoPaymentMethodModal from "./RecurringTopUpNoPaymentMethodModal";

const { React } = cssns("RecurringTopUps");

interface RecurringTopUpsStateProps {
	user?: User;
	showAlertMessage: boolean;
	currentRecurringTopUps: Array<RecurringTopUpProductType>;
	selectedCurrency: string;
	modelForAdd: RecurringTopUpModelType;
	activePaymentMethods: Array<CustomerPaymentMethod>;
}

interface RecurringTopUpsActionProps {
	actions: {
		getPaymentMethods: (customerAccountId: string) => void;
		historyPush: (newLocation: string) => void;
	};
}

interface RecurringTopUpsState {
	showEditModal: boolean;
	showAddModal: boolean;
	showNoPaymentModal: boolean;
	model: RecurringTopUpModelType;
}

type RecurringTopUpsProps = RecurringTopUpsStateProps & RecurringTopUpsActionProps;

class RecurringTopUps extends Component<RecurringTopUpsProps, RecurringTopUpsState> {
	static displayName = "RecurringTopUps";
	static contextTypes = contextTypesValidationMap;

	constructor(props: RecurringTopUpsProps, context: ContextType) {
		super(props, context);

		this.state = {
			showEditModal: false,
			showAddModal: false,
			showNoPaymentModal: false,
			model: {},
		};
	}

	componentWillMount(): void {
		if (this.props.user) {
			const accountId = this.props.user.attributes ? this.props.user.attributes.customerAccountId : this.props.user.customerAccountId;
			this.props.actions.getPaymentMethods(accountId!);
		}
	}

	generateRows = (recurringTopUpProducts: Array<RecurringTopUpProductType>): ReactNode => {
		const { selectedCurrency, activePaymentMethods } = this.props;
		return (
			<tbody>
				{recurringTopUpProducts.map((recurringTopUpProduct: RecurringTopUpProductType, idx: number) => {
					const { subscription, characteristicsValue, paymentMethod } = recurringTopUpProduct;
					const { type, thresholdValue, amount } = characteristicsValue;
					const msisdnWithSeparatedAreaCode = subscription && MsisdnUtil.getMsisdnWithSeparatedAreaCode(subscription);
					const creditCardPresentation = paymentMethod && PaymentUtil.getCreditCardPresentation(paymentMethod);
					const cardType = creditCardPresentation && creditCardPresentation.cardType;
					const cardEnding = creditCardPresentation && creditCardPresentation.cardEnding;

					return (
						<tr key={`RecurringTopUps-table-row-${idx}`}>
							<td className="key-value">
								<div className="key">
									<FormattedMessage {...messages.phoneNumberHeader} />
								</div>
								{msisdnWithSeparatedAreaCode}
							</td>
							<td className="key-value">
								<div className="key">
									<FormattedMessage {...messages.topUpTypeHeader} />
								</div>
								{type}
							</td>
							<td className="key-value">
								<div className="key">
									<FormattedMessage {...messages.amountHeader} />
								</div>
								<div className="right-align">
									<OcCurrency cost={Number(amount)} currency={selectedCurrency} round={true} ignoreSpaces={true} />
								</div>
							</td>
							<td className="key-value">
								<div className="key">
									<FormattedMessage {...messages.thresholdHeader} />
								</div>
								<div className="right-align">
									<OcCurrency cost={Number(thresholdValue)} currency={selectedCurrency} round={true} ignoreSpaces={true} />
								</div>
							</td>
							<td className="key-value">
								<div className="key">
									<FormattedMessage {...messages.paymentMethodHeader} />
								</div>
								<span className="payment-method-col mobile-view-payment-method">
									{creditCardPresentation && <FormattedMessage {...messages.cardDescription} values={{ cardType, cardEnding }} />}
									{!creditCardPresentation && (
										<span
											className="fake-btn-link"
											onClick={() => {
												this.openPaymentMethodTabOrEditModal(activePaymentMethods, recurringTopUpProduct);
											}}
										>
											<i className="fa fa-exclamation-circle add-payment-method-icon" />
											<span className="add-payment-method-link">
												<FormattedMessage {...messages.addPaymentMethod} />
											</span>
										</span>
									)}
								</span>
							</td>
							<td className="key-value">
								<div className="right-align">{this.renderEditButton(idx, recurringTopUpProduct)}</div>
							</td>
						</tr>
					);
				})}
			</tbody>
		);
	};

	updateStateWith = (model: RecurringTopUpModelType, showEditModal: boolean) => {
		this.setState({
			model,
			showEditModal: showEditModal,
		});
	};

	openPaymentMethodTabOrEditModal(activePaymentMethods: CustomerPaymentMethod[], recurringTopUpProduct: RecurringTopUpProductType) {
		if (activePaymentMethods.length === 0) {
			this.props.actions.historyPush(commonDigitalLifeRoutes.DIGITAL_LIFE_PAYMENT_METHODS.createLink());
		} else if (activePaymentMethods.length > 0) {
			const model = RecurringTopUpUtil.getModelForEdit(recurringTopUpProduct);
			this.updateStateWith(model, true);
		}
	}

	renderEditButton(idx: number, recurringTopUpProduct: RecurringTopUpProductType): ReactNode {
		const model = RecurringTopUpUtil.getModelForEdit(recurringTopUpProduct);
		return (
			<span className="active-span btn-link" id={`RecurringTopUps-edit-button-${idx}`} onClick={() => this.updateStateWith(model, true)}>
				<i className="fa fa-pencil-alt btn-icon-left" />
				<FormattedMessage {...messages.edit} />
			</span>
		);
	}

	renderAddButton = (): ReactNode => {
		const { modelForAdd, currentRecurringTopUps, activePaymentMethods } = this.props;
		const isRecurringTopUpsNotEmptyList = currentRecurringTopUps && currentRecurringTopUps.length > 0;
		const isPaymentExist = activePaymentMethods && activePaymentMethods.length > 0;
		return (
			<OcButton
				buttonType={OcButtonType.PRIMARY}
				outline
				className="add-button"
				id="RecurringTopUps-add-new-top-up-button"
				onClick={() => {
					this.setState({
						model: modelForAdd,
						showAddModal: isPaymentExist,
						showNoPaymentModal: !isPaymentExist,
					});
				}}
			>
				{isRecurringTopUpsNotEmptyList ? <FormattedMessage {...messages.addNewTopUp} /> : <FormattedMessage {...messages.addTopUp} />}
			</OcButton>
		);
	};

	getTableHeader(message: FormattedMessageDescriptor, alignRight: boolean = false, additionalClassNames: string = ""): ReactNode {
		return (
			<th
				className={classnames({
					"right-align": alignRight,
					[additionalClassNames]: additionalClassNames,
				})}
			>
				<FormattedMessage {...message} />
			</th>
		);
	}

	closeEditModal = (): void => {
		this.setState({
			showEditModal: false,
			model: {},
		});
	};
	closeAddModal = (): void => {
		this.setState({
			showAddModal: false,
			model: {},
		});
	};
	closeNoPaymentModal = (): void => {
		this.setState({
			showNoPaymentModal: false,
		});
	};
	render(): ReactNode {
		const { currentRecurringTopUps, showAlertMessage } = this.props;
		const { showEditModal, showAddModal, showNoPaymentModal, model } = this.state;
		// TODO: implement: hide alert
		return (
			<div className="RecurringTopUps">
				{currentRecurringTopUps && currentRecurringTopUps.length > 0 ? (
					<>
						<header className="w-view-header">
							<h3>
								<FormattedMessage {...messages.recurringTopUpList} />
							</h3>
							{this.renderAddButton()}
						</header>
						{showAlertMessage && (
							<Alert bsStyle="success" onDismiss={() => {}}>
								<FormattedHTMLMessage {...messages.changesHaveBeenMadeSuccessfully} />
							</Alert>
						)}
						<div className="content">
							<table className="table">
								<thead>
									<tr>
										{this.getTableHeader(messages.phoneNumberHeader)}
										{this.getTableHeader(messages.topUpTypeHeader)}
										{this.getTableHeader(messages.amountHeader, true)}
										{this.getTableHeader(messages.thresholdHeader, true)}
										{this.getTableHeader(messages.paymentMethodHeader, false, "payment-method-col")}
										<th />
									</tr>
								</thead>
								{currentRecurringTopUps.length > 0 && this.generateRows(currentRecurringTopUps)}
							</table>
						</div>
					</>
				) : (
					<RecurringTopUpsEmptyList addProductToBasket={this.renderAddButton} />
				)}
				{showEditModal && <RecurringTopUpEditModal show={true} closeModal={this.closeEditModal} initialModel={model} />}
				{showAddModal && <RecurringTopUpAddModal show={true} closeModal={this.closeAddModal} initialModel={model} />}
				{showNoPaymentModal && <RecurringTopUpNoPaymentMethodModal show={true} closeModal={this.closeNoPaymentModal} />}
			</div>
		);
	}
}

export default RecurringTopUps;
export { RecurringTopUpsState, RecurringTopUpsStateProps, RecurringTopUpsActionProps, RecurringTopUpsProps, RecurringTopUps };
