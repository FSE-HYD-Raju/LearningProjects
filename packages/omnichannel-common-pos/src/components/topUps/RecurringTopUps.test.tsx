import * as React from "react";

import { default as RecurringTopUps, RecurringTopUpsProps } from "./RecurringTopUps";
import { Agreement, User } from "../../redux/types";
import { SimpleDataMock, shallowWithContext, mountWithContext, mockRouterProps, default as TestUtils } from "../../testUtils";

describe("RecurringTopUps", () => {
	let minProps: RecurringTopUpsProps;
	let agreements: Array<Agreement>;
	let context: any;

	beforeAll(() => {
		minProps = {
			...mockRouterProps,
			currentRecurringTopUps: [
				{
					subscription: "1234567890",
					agreementId: "1",
					productId: "product-id",
					productOfferingId: "po-id",
					characteristicsValue: {
						thresholdValue: "10",
						amount: "13",
						monthlyLimit: "50",
						type: "Threshold",
						customerPaymentMethod: "1"
					},
				},
			],
			showAlertMessage: false,
			activePaymentMethods: [],
			modelForAdd: {},
			selectedCurrency: "EUR",
			user: ({
				customerAccountId: "dummy-id",
			} as any) as User,
			actions: {
				getPaymentMethods: jest.fn(),
				historyPush: jest.fn(),
			},
		};
		agreements = ([
			{
				attributes: {
					products: [
						{
							realizingResources: [
								{
									type: "MSISDN",
									lifeCycleStatus: "ACTIVE",
									primaryId: "+1234567890",
								},
							],
							childProducts: [
								{
									lifeCycleStatus: "TERMINATED",
									characteristics: {
										CH_TopUp_Type: "Threshold",
										CH_Threshold_Value: "10",
										CH_TopUp_Amount: "13",
										CH_Monthly_TopUp_Limit: "50",
									},
								},
								{
									lifeCycleStatus: "ACTIVE",
									characteristics: {
										CH_TopUp_Type: "Threshold",
										CH_Threshold_Value: "25",
										CH_TopUp_Amount: "25",
										CH_Monthly_TopUp_Limit: "100",
									},
								},
								{
									lifeCycleStatus: "TERMINATED",
									characteristics: {
										CH_TopUp_Type: "Threshold",
										CH_Threshold_Value: "5",
										CH_TopUp_Amount: "6",
										CH_Monthly_TopUp_Limit: "25",
									},
								},
							],
						},
					],
				},
			},
		] as any) as Array<Agreement>;
		context = SimpleDataMock.getConsulContextMock();
		context.flux.actions = {
			SalesActions: {
				initializeProductReplace: jest.fn(),
			},
			BasketActions: {
				addProductToBasket: jest.fn(),
			},
		};
	});

	const expectInputEquals = (modalContent: any, fieldId: string, shouldBe: any, inputType = "input") => {
		const value = modalContent
			.find(fieldId)
			.at(0)
			.find(inputType)
			.props().value;
		expect(value).toBe(shouldBe);
	};

	it("succeeds at shallow mount with min props", () => {
		const wrapper = shallowWithContext(<RecurringTopUps {...minProps} />);
		expect(wrapper).toMatchSnapshot();
	});

	it("succeeds at deep mount with min props", () => {
		mountWithContext(<RecurringTopUps {...minProps} />, { context });
	});

	it("should render empty table if there are no agreements", () => {
		const props = {
			...minProps,
			currentRecurringTopUps: [],
		};
		const wrapper = mountWithContext(<RecurringTopUps {...props} />, { context });
		expect(wrapper.find("tbody").find("tr").length).toBe(0);
	});

	it("should render table rows for agreements with top-ups", () => {
		const props = {
			...minProps,
			agreements,
		};
		const wrapper = mountWithContext(<RecurringTopUps {...props} />, { context });
		expect(wrapper.find("tbody").find("tr").length).toBe(1);
	});

	it("should show modal after click on edit (threshold top-up)", () => {
		const props = {
			...minProps,
			agreements,
			paymentMethods: [
				{
					id: "dummy-payment",
					attributes: {
						name: "Credit Card",
					},
				},
			],
		};

		const wrapper = shallowWithContext(<RecurringTopUps {...props} />, { context });
		wrapper
			.find("#RecurringTopUps-edit-button-0")
			.at(0)
			.simulate("click");
		const modal = wrapper.find("RecurringTopUpEditModal");
		expect(modal).not.toBeUndefined();
	});

	// same test as before. Should be another test for modal
	xit("should show modal after click on edit (monthly top-up)", () => {
		const agr = ([
			{
				attributes: {
					products: [
						{
							realizingResources: [
								{
									type: "MSISDN",
									lifeCycleStatus: "ACTIVE",
									primaryId: "+1234567890",
								},
							],
							childProducts: [
								{
									lifeCycleStatus: "TERMINATED",
									characteristics: {
										CH_TopUp_Type: "Threshold",
										CH_Threshold_Value: "10",
										CH_TopUp_Amount: "13",
										CH_Monthly_TopUp_Limit: "50",
									},
								},
								{
									lifeCycleStatus: "ACTIVE",
									characteristics: {
										CH_TopUp_Type: "Time",
										CH_TopUp_Amount: "10",
										CH_Time_Interval: "month",
										CH_Time_Interval_Count: "1",
									},
								},
								{
									lifeCycleStatus: "TERMINATED",
									characteristics: {
										CH_TopUp_Type: "Threshold",
										CH_Threshold_Value: "5",
										CH_TopUp_Amount: "6",
										CH_Monthly_TopUp_Limit: "25",
									},
								},
							],
						},
					],
				},
			},
		] as any) as Array<Agreement>;
		const props = {
			...minProps,
			agreements: agr,
			paymentMethods: [
				{
					id: "dummy-payment",
					attributes: {
						name: "Credit Card",
					},
				},
			],
		};

		const wrapper = shallowWithContext(<RecurringTopUps {...props} />, { context });
		wrapper
			.find("#RecurringTopUps-edit-button-0")
			.at(0)
			.simulate("click");

		const modal = wrapper.find("RecurringTopUpEditModal");
		const modalContent = TestUtils.getModalContents(modal, context);

		expect(
			modalContent
				.find("select")
				.at(0)
				.props().value
		).toBe("+1234567890");

		expect(
			modalContent
				.find("input[name='recurringTopUp']")
				.at(2)
				.props().checked
		).toBeTruthy();

		expectInputEquals(modalContent, "#RecurringTopUpEditForm-monthly-top-up-amount-field", 10);
	});
});
