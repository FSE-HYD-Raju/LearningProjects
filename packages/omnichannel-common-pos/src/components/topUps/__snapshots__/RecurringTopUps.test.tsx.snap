// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecurringTopUps succeeds at shallow mount with min props 1`] = `
<div
  className="RecurringTopUps"
>
  <header
    className="w-view-header"
  >
    <h3>
      <FormattedMessageWrapper
        defaultMessage="Recurring top-up"
        description="Recurring top-up list title"
        id="recurring-top-up-list-title"
        wrapperType="wrapper"
      />
    </h3>
    <OcButton
      buttonType="PRIMARY"
      className="RecurringTopUps-add-button"
      id="RecurringTopUps-add-new-top-up-button"
      onClick={[Function]}
      outline={true}
    >
      <FormattedMessageWrapper
        defaultMessage="Add new recurring top-up"
        description="Recurring top-ups top-up button message"
        id="recurring-top-ups-top-up-button"
        wrapperType="wrapper"
      />
    </OcButton>
  </header>
  <div
    className="RecurringTopUps-content"
  >
    <table
      className="table"
    >
      <thead>
        <tr>
          <th
            className=""
          >
            <FormattedMessageWrapper
              defaultMessage="Phone number"
              description="digital-life, recurring top-ups table header phone number"
              id="recurring-top-ups-table-header-phone-number"
              wrapperType="wrapper"
            />
          </th>
          <th
            className=""
          >
            <FormattedMessageWrapper
              defaultMessage="Top-up type"
              description="digital-life, recurring top-ups table header top-up type"
              id="recurring-top-ups-table-header-top-up-type"
              wrapperType="wrapper"
            />
          </th>
          <th
            className="RecurringTopUps-right-align"
          >
            <FormattedMessageWrapper
              defaultMessage="Amount"
              description="digital-life, recurring top-ups table header amount"
              id="recurring-top-ups-table-header-amount"
              wrapperType="wrapper"
            />
          </th>
          <th
            className="RecurringTopUps-right-align"
          >
            <FormattedMessageWrapper
              defaultMessage="Threshold"
              description="digital-life, recurring top-ups table header threshold value"
              id="recurring-top-ups-table-header-threshold-value"
              wrapperType="wrapper"
            />
          </th>
          <th
            className="RecurringTopUps-payment-method-col"
          >
            <FormattedMessageWrapper
              defaultMessage="Payment-method"
              description="digital-life, recurring top-ups table header payment method"
              id="recurring-top-ups-table-header-payment-method"
              wrapperType="wrapper"
            />
          </th>
          <th />
        </tr>
      </thead>
      <tbody>
        <tr
          key="RecurringTopUps-table-row-0"
        >
          <td
            className="RecurringTopUps-key-value"
          >
            <div
              className="RecurringTopUps-key"
            >
              <FormattedMessageWrapper
                defaultMessage="Phone number"
                description="digital-life, recurring top-ups table header phone number"
                id="recurring-top-ups-table-header-phone-number"
                wrapperType="wrapper"
              />
            </div>
            1234567890
          </td>
          <td
            className="RecurringTopUps-key-value"
          >
            <div
              className="RecurringTopUps-key"
            >
              <FormattedMessageWrapper
                defaultMessage="Top-up type"
                description="digital-life, recurring top-ups table header top-up type"
                id="recurring-top-ups-table-header-top-up-type"
                wrapperType="wrapper"
              />
            </div>
            Threshold
          </td>
          <td
            className="RecurringTopUps-key-value"
          >
            <div
              className="RecurringTopUps-key"
            >
              <FormattedMessageWrapper
                defaultMessage="Amount"
                description="digital-life, recurring top-ups table header amount"
                id="recurring-top-ups-table-header-amount"
                wrapperType="wrapper"
              />
            </div>
            <div
              className="RecurringTopUps-right-align"
            >
              <OcCurrency
                cost={13}
                currency="EUR"
                ignoreSpaces={true}
                round={true}
              />
            </div>
          </td>
          <td
            className="RecurringTopUps-key-value"
          >
            <div
              className="RecurringTopUps-key"
            >
              <FormattedMessageWrapper
                defaultMessage="Threshold"
                description="digital-life, recurring top-ups table header threshold value"
                id="recurring-top-ups-table-header-threshold-value"
                wrapperType="wrapper"
              />
            </div>
            <div
              className="RecurringTopUps-right-align"
            >
              <OcCurrency
                cost={10}
                currency="EUR"
                ignoreSpaces={true}
                round={true}
              />
            </div>
          </td>
          <td
            className="RecurringTopUps-key-value"
          >
            <div
              className="RecurringTopUps-key"
            >
              <FormattedMessageWrapper
                defaultMessage="Payment-method"
                description="digital-life, recurring top-ups table header payment method"
                id="recurring-top-ups-table-header-payment-method"
                wrapperType="wrapper"
              />
            </div>
            <span
              className="RecurringTopUps-payment-method-col RecurringTopUps-mobile-view-payment-method"
            >
              <span
                className="RecurringTopUps-fake-btn-link"
                onClick={[Function]}
              >
                <i
                  className="fa fa-exclamation-circle RecurringTopUps-add-payment-method-icon"
                />
                <span
                  className="RecurringTopUps-add-payment-method-link"
                >
                  <FormattedMessageWrapper
                    defaultMessage="Add payment method"
                    description="Recurring top-up add payment method"
                    id="recurring-top-up-add-payment-method"
                    wrapperType="wrapper"
                  />
                </span>
              </span>
            </span>
          </td>
          <td
            className="RecurringTopUps-key-value"
          >
            <div
              className="RecurringTopUps-right-align"
            >
              <span
                className="RecurringTopUps-active-span btn-link"
                id="RecurringTopUps-edit-button-0"
                onClick={[Function]}
              >
                <i
                  className="fa fa-pencil-alt btn-icon-left"
                />
                <FormattedMessageWrapper
                  defaultMessage="Edit"
                  description="Recurring top-ups edit button"
                  id="recurring-top-ups-top-edit-button"
                  wrapperType="wrapper"
                />
              </span>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
